src/main/java/com/joyent/manta/monitor/MantaMonitorJerseyServer.java

```diff
@@ -141,10 +219,35 @@ private void configureServer() {
             httpsConfig.addCustomizer(src);
 
             SslContextFactory sslContextFactory = new SslContextFactory();
-            sslContextFactory.setKeyStorePath(System.getenv("KEYSTORE_PATH"));
+
+            String userHomePath = System.getProperty("user.home");
+
+            URI keystoreURI = getURIFromString(System.getenv("KEYSTORE_PATH"));
+
+            if ("manta".equals(keystoreURI.getScheme())) {
+                // Make the keystore file available for the sslContextFactory, by
+                // reading the remote file and writing it to a file on the host file system.
+                File keystoreTargetFile = new File(userHomePath + "/keystore");
+                writeInputStreamToTarget(keystoreURI, keystoreTargetFile);
+                sslContextFactory.setKeyStorePath(keystoreTargetFile.getPath());
+            } else {
+                // Read the keystore directly from the local file system
+                sslContextFactory.setKeyStorePath(System.getenv("KEYSTORE_PATH"));
+            }
+
+            URI trustStoreURI = getURIFromString(System.getenv("TRUSTSTORE_PATH"));
+            if ("manta".equals(trustStoreURI.getScheme())) {
+                // Make the truststore file available for the sslContextFactory, by
+                // reading the remote file and writing it to a file on the host file system.
+                File trustStoreTargetFile = new File(userHomePath + "/truststore");
```

Comment:
Does this mean we look for the truststore at `$HOME/truststore`?

Reply:
Yes. The truststore file after being fetched from the manta store is locally stored under ```System.getProperty("user.home")/truststore``` so that it will be available when trying to set the truststorepath.

Reply:
I suggest making a subdir to home like `$HOME/.manta-monitor/truststore`.

Also, use `File.separator` instead of hardcoding slashes in Java apps.

Reply:
Added the changes.